### ISSUE TITLE
'-llw', '--l_loss_weight'

### ISSUE THREAD

**Issue by xxx2974** [huanghoujing/AlignedReID-Re-Production-Pytorch#62]
'-llw', '--l_loss_weight'：我想问一下这个参数在训练过程中为什么要设置成0呢？训练时不是应该局部距离与全局距离都用吗？

    # Only for training set.
    parser.add_argument('--resize_h_w', type=eval, default=(224, 224)) #256,128
    parser.add_argument('--crop_prob', type=float, default=0)
    parser.add_argument('--crop_ratio', type=float, default=1)
    parser.add_argument('--ids_per_batch', type=int, default=40)  #32
    parser.add_argument('--ims_per_id', type=int, default=4)

    parser.add_argument('--log_to_file', type=str2bool, default=True)
    parser.add_argument('--normalize_feature', type=str2bool, default=True)
    parser.add_argument('--local_dist_own_hard_sample',
                        type=str2bool, default=False)
    parser.add_argument('-gm', '--global_margin', type=float, default=0.5) #0.3
    parser.add_argument('-lm', '--local_margin', type=float, default=0.5)  #0.3
    parser.add_argument('-glw', '--g_loss_weight', type=float, default=1.)
    parser.add_argument('-llw', '--l_loss_weight', type=float, default=1.)#0.
    parser.add_argument('-idlw', '--id_loss_weight', type=float, default=0.)

**Comment by huanghoujing** [comment]
你好，这个是可以设成1的。这里设成0只是为了对应`Triplet Loss`和`Triplet Loss + Mutual Loss`的分数。

**Comment by xxx2974** [comment]
额，谢谢，不过我有些不太明白，这个是train.py文件中的参数，是没有用协同损失的。
（1）llw该参数使用与否和协同学习中的损失有关系吗？？？
（2）我理解的llw这个参数的意思是不是代表局部损失的权重？我看论文里在训练阶段是有用到这个局部损失的呀？？？？

**Comment by huanghoujing** [comment]
是的，llw是local distance的损失，如果跟论文一样的设置，就需要把这个设为1。但是这个代码也可以让你把这些参数设成其它的，进行各种实验。

**Comment by xxx2974** [comment]
我把llw设为1训练过，但是结果很差，是为什么呢？

**Comment by huanghoujing** [comment]
我跑的实验记录在[这个Excel文件](https://github.com/huanghoujing/AlignedReID-Re-Production-Pytorch/blob/master/AlignedReID-Scores.xlsx)中。加了llw，并且测试也用上local distance，是有提升的，你的这个问题我解决不了。。。
